### PR TITLE
Doc: improve README

### PR DESCRIPTION
**Pull request by yunwei37**
- move build to doc/build.md
- move examples to examples/README.md
- add go examples description
- add rust runtime description

Any more suggestions? The Chinese documents improvement should be in another PR.

### CODE REVIEW

**Comment by yunwei37** [comment]
shall we keep an example output in readme， to show the usage of this program?

File: README.md [hunk 1]
```diff
@@ -43,199 +43,30 @@ We have proposed a new WASI issue [wasi-bpf](https://github.com/WebAssembly/WASI
...
-
-See [examples/sockfilter](examples/sockfilter) for more details.
-
-### C example: Sockops
-
-`sockops` add the pid int tcp option in syn packet.
-
-See [examples/sockops](examples/sockops) for more details.
-
 ## build the runtime
```
Comment (septs) [inline README.md:55]:
```suggestion
## Build the runtime
```

File: README.md [hunk 2]
```diff
@@ -43,199 +43,30 @@ We have proposed a new WASI issue [wasi-bpf](https://github.com/WebAssembly/WASI
...
+- A C/C++ runtime example, which is a minimal runtime based on WAMR. see [runtime/cpp](../runtime/cpp) for more details.
+- A Rust runtime example, which is a more complex runtime based on Wasmtime. see [runtime/rust](../runtime/rust) for more details.
 
-You may refer to [CI](.github/workflows/c-cpp.yml) for more details on how
-to build and run the examples.
+The runtime can be built as a library or a standalone executable. see [docs/build.md](docs/build.md) to build the runtimes.
 
 ## LICENSE
 
 MIT
```
Comment (septs) [inline README.md:66]:
```suggestion
[MIT LICENSE](LICENSE)
```

File: docs/build.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# build the runtime and examples
```
Comment (septs) [inline docs/build.md:1]:
```suggestion
# Build the runtime and examples
```

File: docs/build.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+We have two types of runtime examples:
+
+- A C/C++ runtime example, which is a minimal runtime based on WAMR. see [runtime/cpp](../runtime/cpp) for more details.
+- A Rust runtime example, which is a more complex runtime based on Wasmtime. see [runtime/rust](../runtime/rust) for more details.
+
+A new runtime is easy to implement with only a few hundred lines of code, in any language, using any wasm runtime or any ebpf user space library.
+
+## build the Cpp minimal runtime based on WAMR
```
Comment (septs) [inline docs/build.md:10]:
```suggestion
## Build the C++ minimal runtime based on WAMR[^wamr]
```

need add wamr abbreviation footnote

File: docs/build.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+You will need `clang`, `libelf` and `zlib` to build the examples,
+package names may vary across distros.
+
+On Ubuntu/Debian, you need:
+
+```shell
+sudo apt install clang libelf1 libelf-dev zlib1g-dev
+```
+
+On CentOS/Fedora, you need:
```
Comment (septs) [inline docs/build.md:31]:
```suggestion
on CentOS / Fedora, you need:
```

File: docs/build.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+git submodule update --init --recursive
+cd runtime/cpp
+```
+
+### Install Dependencies
+
+You will need `clang`, `libelf` and `zlib` to build the examples,
+package names may vary across distros.
+
+On Ubuntu/Debian, you need:
```
Comment (septs) [inline docs/build.md:25]:
```suggestion
on Ubuntu/Debian, you need:
```

File: docs/build.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+### Build runtime as a library
+
+```sh
+make build-lib
+```
+
+You may refer to [CI](.github/workflows/c-cpp.yml) for more details on how
+to build and run the examples.
+
+## build the Rust runtime based on Wasmtime
```
Comment (septs) [inline docs/build.md:55]:
```suggestion
## Build the Rust runtime based on Wasmtime
```

Comment (ocfox) [inline docs/build.md:55]:
```suggestion
## Build the Rust runtime based on Wasmtime
```

File: docs/build.md (hunk 4 above)
Comment (ocfox) [inline docs/build.md:10]:
```suggestion
## Build the Cpp minimal runtime based on WAMR
```

File: docs/build.md (hunk 3 above)
Comment (ocfox) [inline docs/build.md:1]:
```suggestion
# Build the runtime and examples
```

File: examples/README.md [hunk 8]
```diff
@@ -7,4 +7,138 @@ Each directory contains one example.
...
+
+See [examples/rust-bootstrap](examples/rust-bootstrap) for more details.
+
+### C example: [runqlat](examples/runqlat)
+
+This program summarizes scheduler run queue latency as a histogram, showing
+how long tasks spent waiting their turn to run on-CPU.
+
+This program summarizes scheduler run queue latency as a histogram, showing
+how long tasks spent waiting their turn to run on-CPU.
```
Comment (ocfox) [inline examples/README.md:73]:
```suggestion
This program summarizes scheduler run queue latency as a histogram, showing
how long tasks spent waiting their turn to run on-CPU.
```

File: README.md [hunk 9]
```diff
@@ -43,199 +43,30 @@ We have proposed a new WASI issue [wasi-bpf](https://github.com/WebAssembly/WASI
...
 
-- [bootstrap](examples/bootstrap) and [runqlat](examples/runqlat) `tracing examples`
-- [rust-bootstrap](examples/rust-bootstrap) `tracing examples`
-- [lsm](examples/lsm) `security example`
+- [bootstrap](examples/bootstrap) and [rust-bootstrap](examples/rust-bootstrap) `tracing examples`
+- [runqlat](examples/runqlat) `tracing examples`
+- [go-execve](examples/go-execve) `tracing examples`
+- [lsm](examples/lsm) and  `security example`
 - [sockfilter](examples/sockfilter) `networking example`
 - [sockops](examples/sockops) `networking example`
```
Comment (ocfox) [inline README.md:53]: Maybe better with the following format?

- [sockops](examples/sockops): Add the pid int tcp option in syn packet.

File: examples/README.md [hunk 10]
```diff
@@ -7,4 +7,135 @@ Each directory contains one example.
...
+
+See [examples/rust-bootstrap](examples/rust-bootstrap) for more details.
+
+### C example: [runqlat](examples/runqlat)
+
+This program summarizes scheduler run queue latency as a histogram, showing
+how long tasks spent waiting their turn to run on-CPU.
+
+```console
+$ sudo ./wasm-bpf runqlat.wasm -h
```
Comment (ocfox) [inline examples/README.md:73]:
```suggestion
$ sudo ./wasm-bpf -h
```

Comment (septs) [reply examples/README.md:73]: if support hashbang？

```suggestion
$ sudo ./runqlat.bpf -h
```

Comment (ocfox) [reply examples/README.md:73]: I misunderstood, this help information is from wasm.
No need to change.

File: README.md (hunk 9 above)
Comment (yunwei37) [reply README.md:53]: Yes! Thank you for notification!

**Pull request merged**